### PR TITLE
Fix/default value button group queries

### PR DESCRIPTION
**Pull request by kwongz**
### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable
- ~~I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

https://github.com/user-attachments/assets/920e9cbf-1853-4f4b-ab5f-b9e474f216bc

### CODE REVIEW

File: packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte [hunk 1]
```diff
@@ -22,6 +22,15 @@
 	/** @type {keyof typeof presets | undefined} */
 	export let preset = undefined;
 
+	// for Tabs styling
+	export let display;
```
Comment (archiewood) [inline packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte:26]: `= 'buttons'` ?

File: packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte [hunk 2]
```diff
@@ -22,6 +22,15 @@
...
+
+	export let defaultValue;
```
Comment (archiewood) [inline packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte:28]: `=undefined`?

**Review by archiewood — CHANGES_REQUESTED** [review]

File: packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte (hunk 1 above)
Comment (kwongz) [reply packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte:26]: How do we feel about display = 'buttons', and a error state that tells the user when display string does not exist

(similar to how we handle when presets do not exist)
image

File: packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte (hunk 2 above)
Comment (kwongz) [reply packages/ui/core-components/src/lib/atoms/inputs/button-group/ButtonGroup.svelte:28]: Sounds good, more definitive, will also add they typescripting
image

**Comment by kwongz** [comment]
@archiewood 

Added the recommended definitions, 
Added error handling and messages

Added error stories
image

Checked evidence app w/ errors
image

**Comment by kwongz** [comment]
Added type to non string errors

more clear that the user put in an array/object

image

image

**Comment by archiewood** [comment]
Seems sensible, decent error messages:

Should use Evidence error components for user facing errors:

- https://github.com/evidence-dev/evidence/blob/next/packages/ui/core-components/src/lib/unsorted/viz/core/ValueError.svelte
- https://github.com/evidence-dev/evidence/blob/next/packages/ui/core-components/src/lib/unsorted/viz/core/ErrorChart.svelte

**Comment by kwongz** [comment]
Added error-chart for error display handling + cleaned up error check

docs:
image

Story:
image

**Comment by kwongz** [comment]
@archiewood Added errorchart + cleanup error check logic

image

**Review by archiewood — APPROVED** [review]
Looks good from my testing.

Added simpler example to the docs, and documented an undocumented prop

**Pull request merged**